src/init.cpp

```diff
@@ -1256,6 +1260,27 @@ bool AppInitMain()
             gArgs.GetArg("-datadir", ""), fs::current_path().string());
     }
 
+    if (g_has_master_datadir) {
+        // LDB does not support readonly access, so we need to copy the entire
+        // chainstate and blocks/index dirs over to the new location, if not found already
+        fs::path path = GetDataDir() / "chainstate";
+        if (!fs::exists(path)) {
```

Comment:
Doesn't this copy action require a protection that the masterdatadir is currently not in use/write? If a Bitcoin Core instance is using this (master) datadir it may result in a corrupted copy?

Reply:
Yes. Master is guaranteed to be safe as no write operation is ever done from the slave, but the opposite is not the case. See https://github.com/bitcoin/bitcoin/pull/13746#issuecomment-407957987 for experiment on corruption. There's no good way to recover, as the slave will have already written data and closed the file before realizing it was truncated.

The question remains how frequent this kind of race condition is, though.